src/native/corehost/apphost/static/CMakeLists.txt

```diff
@@ -69,11 +69,16 @@ if(CLR_CMAKE_TARGET_WIN32)
     add_compile_definitions(UNICODE)
     list(APPEND SOURCES
         ../apphost.windows.cpp
+		../appcontainer_helpers.cpp
         ${CLR_SRC_NATIVE_DIR}/libs/Common/delayloadhook_windows.cpp
     )
 
     list(APPEND HEADERS
-        ../apphost.windows.h)
+        ../apphost.windows.h
+		../appcontainer_helpers.h
+	)
+	
+	set_source_files_properties(../appcontainer_helpers.cpp PROPERTIES COMPILE_FLAGS "/std:c++17 /O1")
```

Comment:
Pulling in C++17 isn't something we want to do in a piece meal fashion. Either we add support for everything or don't. I'm not sure we are ready to do that move yet.